### DISCUSSION TITLE
Adding setCookie capability to xk6-browser

### DISCUSSION THREAD

**Discussion by krithi0503** [grafana/xk6-browser#329] [Q&A]
As it stands, setCookie() method to add cookies to browserContext for playwright is not supported on xk6 browser. Can this capability be added? Or if not is there another alternate way to achieve this?

**Comment by imiric** [comment]
Hi there! This is on our short-term roadmap (see #5), but there are other issues we're prioritizing first. Feel free to thumbs-up and follow that issue for updates.

In the meantime, you can set cookies via [`document.cookie`](https://developer.mozilla.org/en-US/docs/web/api/document/cookie) in a `page.evaluate()` call, but this will only work per document and probably won't be useful for authentication.

**Reply by krithi0503** [reply]
Hi, We tried something like this but it did not work. Not sure if there are any format/function call issues. There is a need to add two cookies to load the application under test. Hence we are trying to add this cookie before doing a page.goTo() 

```
    page.evaluate(() => {
        document.cookie = "cookie1=value1";
        document.cookie = "cookie2=value2";
    });
```

Error 
```
ERRO[0001] error calling evaluate: frame cannot evaluate: cannot call function on expression ("() => document.cookie = \"cookie1=value1\"\n//# sourceURL=__xk6_browser_evaluation_script__\n") in execution context (1) in frame (B711ACBB34F66DB3C5CCE8B9DFA9C918) with session (F1E712BBD3B9B6A7641A5003A83ECD45): exception "Uncaught" (0:23): DOMException: Failed to set the 'cookie' property on 'Document': Access is denied for this document.
    at __xk6_browser_evaluation_script__:1:24map[code:%!s(float64=18) message:Failed to set the 'cookie' property on 'Document': Access is denied for this document. name:SecurityError stack:accessor]
        at reflect.methodValueCall (native)
        at file:///C:/Users/Kiruthika.Ponnusamy/K6_Load/load_tests/sampleBrowser.js:30:4(26)
        at native  executor=per-vu-iterations scenario=default source=stacktrace
ERRO[0001] err:read tcp 127.0.0.1:54319->127.0.0.1:54318: wsarecv: An existing connection was forcibly closed by the remote host.  category="Connection:handleIOError" elapsed="0 ms" goroutine=79

```

Do Let us know if there is anything we could try!

**Reply by imiric** [reply]
Yeah, sorry, there's no workaround for your use case until #5 is implemented.
